### PR TITLE
Switch Docs generation to Xcode 13.1

### PR DESCRIPTION
**Pull request by pjleonard37**
Jazzy is not reliable with Xcode 14 and generates broken docs with missing symbols. This PR moves `build-docs` 
to xcode-sdk-min (currently Xcode 13.1) to work around this. 

xcode-latest is currently Xcode 14.1

## Pull request checklist:
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. Put tests in correct [Test Plan](https://github.com/mapbox/mapbox-maps-ios/tree/main/Tests/TestPlans) (Unit, Integration, All)
   - [ ] If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` first and then port to `v10.[version]` release branch.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Review by ZiZasaurus — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -981,7 +981,7 @@ jobs:
       - slack-notify-failure-if-any
 
   build-docs:
-    executor: xcode-latest
+    executor: xcode-sdk-min
```
Comment (OdNairy) [inline .circleci/config.yml:984]: Can we fix the 13.4.1 here for now?

Comment (pjleonard37) [reply .circleci/config.yml]: 👍  -- yes, updated in latest commit

File: .circleci/config.yml [hunk 2]
```diff
@@ -981,7 +981,7 @@ jobs:
       - slack-notify-failure-if-any
 
   build-docs:
-    executor: xcode-latest
+    executor: 13.4.1
```
Comment (OdNairy) [inline .circleci/config.yml:984]: We don't have a 13.4.1 executor. Please, check the CIrcleCI doc to assign corresponding macOS/Xcode image

Comment (pjleonard37) [reply .circleci/config.yml]: Updated in latest

File: .circleci/config.yml [hunk 3]
```diff
@@ -981,7 +981,9 @@ jobs:
       - slack-notify-failure-if-any
 
   build-docs:
-    executor: xcode-latest
+    executor:
```
Comment (OdNairy) [inline .circleci/config.yml:984]: You don't need an executor root key. You can assign Xcode value to the `macos` key

Comment (pjleonard37) [reply .circleci/config.yml:984]: Do you mean like this?
``` 
build-docs: 
  macos:
    xcode: 13.4.1
```

Comment (OdNairy) [reply .circleci/config.yml:984]: yes

Comment (pjleonard37) [reply .circleci/config.yml]: Great -- updated in latest

**Review by OdNairy — APPROVED** [review]

**Pull request merged**